source/common/http/http1/codec_impl.cc

```diff
@@ -57,6 +57,43 @@ HeaderKeyFormatterPtr formatter(const Http::Http1Settings& settings) {
 
   return nullptr;
 }
+
+/**
+ * Determines whether Envoy supports the encodings provided for the transfer-encoding header. Envoy
+ * currently only supports chunked and identity. Per RFC 7230 Section 3.3.3 when encodings are
+ * chained then chunked must be the last encoding, so possible values are: chunked identity,chunked
+ * identity, chunked
+ * etc.
+ *
+ * @param encoding The transfer-encoding value from the HTTP message
+ * @return whether the encoding only contains supported encodings.
+ */
+bool IsSupportedTransferEncoding(absl::string_view encoding) {
+  const std::string& chunked = Headers::get().TransferEncodingValues.Chunked;
+  if (absl::EqualsIgnoreCase(encoding, chunked)) {
+    return true;
+  }
+
+  // The only other encoding we support is identity when chained with chunked.
```

Comment:
What about "transfer-encoding: identity"?
That seemed allowed by the original version.

Reply:
It is not allowed by HTTP specification anymore although it might have been tolerated in previous versions of the specification. The whole business with the Transfer-Encoding strictness is that it is possible to do nasty HTTP smuggling attacks when the parser is too lenient.

Reply:
To be more precise: from [RFC 7230 3.3.3](https://tools.ietf.org/html/rfc7230#section-3.3.3)

```
       If a Transfer-Encoding header field is present in a response and
       the chunked transfer coding is not the final encoding, the
       message body length is determined by reading the connection until
       it is closed by the server.  If a Transfer-Encoding header field
       is present in a request and the chunked transfer coding is not
       the final encoding, the message body length cannot be determined
       reliably; the server MUST respond with the 400 (Bad Request)
       status code and then close the connection.
```

Reply:
Thanks.